pilot/pkg/xds/xds_test.go

```diff
@@ -543,11 +544,14 @@ spec:
 					fakeOpts.MeshConfig = &meshConfig
 					s := NewFakeDiscoveryServer(t, fakeOpts)
 					for clusterID := range want {
-						p := &model.Proxy{Metadata: &model.NodeMetadata{ClusterID: clusterID}}
-						eps := xdstest.ExtractLoadAssignments(s.Endpoints(p))[tt.serviceCluster]
-						if want := want[clusterID]; !listEqualUnordered(eps, want) {
-							t.Errorf("got %v but want %v for %s", eps, want, clusterID)
-						}
+						retry.UntilSuccessOrFail(t, func() error {
```

Comment:
We should revert his. Test should not need retry around XDS generation, it is intended to be handledby the NewFakeDiscoveryServer